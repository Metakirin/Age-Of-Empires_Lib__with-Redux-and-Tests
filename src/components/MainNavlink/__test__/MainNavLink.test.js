import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { paths } from '../../../config'
import { MainNavLink } from '..';
// import { }

configure({ adapter: new Adapter() });

describe('Should render correctly "MainNavLink"', () => {
    const mockFn = jest.fn();

    const props = {
        getCivilizationsAction: mockFn,
        getUnitsAction: mockFn,
        getStructuresAction: mockFn,
        getTechnologiesAction: mockFn,
        title: 'test title',
        path: '/civilizations'
    }
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<MainNavLink {...props} />)
    });
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1);
    });
    it('component have children div with "menu_link" class', () => {
        expect(wrapper.find('.menu_link')).toHaveLength(1);
    });
    it('component have children "NavLink"', () => {
        expect(wrapper.find('NavLink')).toHaveLength(1);
    });
    it('"NavLink" should call onClick when clicking', () => {
        wrapper.find('NavLink').simulate('click');
        expect(props.getCivilizationsAction).toHaveBeenCalledTimes(1);
    });
    describe('getUnitsAction', () => {
        it('"NavLink" should call getUnitsAction when clicking after changin props', () => {
            wrapper.setProps({
                path: '/units'
            })
            wrapper.find('NavLink').simulate('click');
            expect(props.getUnitsAction).toHaveBeenCalledTimes(1);
        });
    })
    describe('111', () => {
        it('"NavLink" should call getStructuresAction when clicking after changin props', () => {
            wrapper.setProps({
                path: '/structures'
            })
            wrapper.find('NavLink').simulate('click');
            expect(props.getStructuresAction).toHaveBeenCalledTimes(1);
        });
    })
    describe('111', () => {
        it('"NavLink" should call getTechnologiesAction when clicking after changin props', () => {
            wrapper.setProps({
                path: '/technologies'
            })
            wrapper.find('NavLink').simulate('click');
            expect(props.getTechnologiesAction).toHaveBeenCalledTimes(1);
        });
    })
})