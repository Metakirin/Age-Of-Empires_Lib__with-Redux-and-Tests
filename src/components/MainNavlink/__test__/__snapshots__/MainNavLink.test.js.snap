// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "MainNavLink" to Match Snapshot 1`] = `
<div
  className="menu_link"
>
  <NavLink
    activeClassName="nav_active"
    exact={true}
    onClick={[Function]}
    to="/civilizations"
  >
    test title
  </NavLink>
</div>
`;
